micro_sam/automatic_segmentation.py

```diff
@@ -226,11 +227,19 @@ def _convert_argval(value):
         parameter_args[i].lstrip("--"): _convert_argval(parameter_args[i + 1]) for i in range(0, len(parameter_args), 2)
     }
 
+    amg = None
+    if args.mode is None:
```

Comment:
You don't need this `if` since you have `amg = None` already in the line before.
Just do `if args.mode is not None` and then what you currently have in Line 234 and following.
(Just a minor thing, but it's more readable.)

Reply:
Ah yeah right. I missed that. I'll fix this. Thanks for spotting!